app/models/fixed_asset.rb

```diff
@@ -202,6 +210,7 @@ def self.state_machine(*args)
     # end
     # end
     depreciate! if @auto_depreciate
+    sale.update_columns(invoiced_at: sold_on.to_datetime) if changes[:sold_on] && sale && !sale.invoice? && sold_on
```

Comment:
Indeed, I wanted to avoid triggering `sale` callbacks since updating a sale updates the fixed asset it's linked to and inversely, which would result in an unnecessary update or even worse in an infinite loop.